src/main/java/seedu/address/logic/LogicManager.java

```diff
@@ -23,20 +24,23 @@
     private final CommandHistory history;
     private final AddressBookParser addressBookParser;
     private final UndoRedoStack undoRedoStack;
+    private final GraphWrapper graphWrapper;
 
     public LogicManager(Model model) {
         this.model = model;
         this.history = new CommandHistory();
         this.addressBookParser = new AddressBookParser();
         this.undoRedoStack = new UndoRedoStack();
+        this.graphWrapper = new GraphWrapper();
+        graphWrapper.buildGraph(model);
```

Comment:
yes the graph is automatically generated when the LogicManager is built i.e. when the application starts. This is because if the first command the user types is one of the commands that do not need to rebuild the graph, like addRelationship, removeTag..., in this case, it will be ideal to show a relationship dynamically built, which means there has to be a previous graph and the relationship is added onto the graph.